XenAdmin/Commands/DisablePvsReadCachingCommand.cs

```diff
@@ -97,8 +97,7 @@ protected virtual void Execute(IList<VM> vms)
         /// <returns></returns>
         private AsyncAction GetAsyncActionForVm(VM vm)
         {
-            var pvsProxy = vm.Connection.Cache.PVS_proxies.FirstOrDefault(p => p.VM.Equals(vm)); // null if no proxy
-
+            var pvsProxy = vm.PvsProxy; 

```

Comment:
Can't we just write in line 67:
```
foreach (var vm in vms)
{
    var pvsProxy = vm.PvsProxy;
    if (pvsProxy != null)
        actions.Add(new PvsProxyDestroyAction(pvsProxy));
}
```
Then we don't need the GetAsyncActionForVm at all - simpler and we gain a few picoseconds by avoiding extra calls and checks.